### PR TITLE
Format code also adding required attribute to those fields that need it.

### PR DESCRIPTION
**Pull request by anthonythorne**
Add basic form field validation, required, and number ranges.
Also formatted the code in the forms directory to be in line with wpphpcs nags.
Updated readme to elaborate on what fields are available on some of the filters.

https://github.com/TheCodeCompany/wp-chargify/issues/114

### CODE REVIEW

File: plugins/wp-chargify/includes/forms/signup.php [hunk 1]
```diff
@@ -8,43 +16,106 @@
...
-	]);
+	$signup_form_fields = new_cmb2_box(
+		[
+			'id'           => $signup_form_id,
+			'object_types' => [ 'chargify_account' ],
+			'hookup'       => false,
+			'save_fields'  => false,
+			'autocomplete' => 'on',
+			'attributes'   => [
+				'class'        => 'test',
```
Comment (BronsonQuick) [inline plugins/wp-chargify/includes/forms/signup.php]: Class `test` seems like a strange class name?

File: plugins/wp-chargify/includes/forms/signup.php [hunk 2]
```diff
@@ -8,43 +16,106 @@
...
-		'hookup'       => false,
-		'save_fields'  => false,
-	]);
+	$signup_form_fields = new_cmb2_box(
+		[
+			'id'           => $signup_form_id,
+			'object_types' => [ 'chargify_account' ],
+			'hookup'       => false,
+			'save_fields'  => false,
+			'autocomplete' => 'on',
```
Comment (BronsonQuick) [inline plugins/wp-chargify/includes/forms/signup.php]: Good catch on the missing `autocomplete`!

File: plugins/wp-chargify/readme.txt [hunk 3]
```diff
@@ -49,16 +49,65 @@ We provide a number of optional filters you can use:
...
 `chargify_generate_password` - Alter the password generation.
+
+== Chargify signup form filters. ==
+
 `chargify_message_details_fields` - ilter the Message Details form field. Use `$signup_form->add_field` and `$signup_form->remove_field` to add and/or remove fields.
 `chargify_form_message_html` - Filter the message html. $html and array of $form_messages that can manipulate the html, with html required as the return value.
+
 `chargify_customer_details_fields` - Filter the Customer Details form field. Use `$signup_form->add_field` and `$signup_form->remove_field` to add and/or remove fields.
+	Available fields by id;
+		'chargify_customer_details_title'
```
Comment (BronsonQuick) [inline plugins/wp-chargify/readme.txt]: There's an extra tab here

File: plugins/wp-chargify/readme.txt [hunk 4]
```diff
@@ -49,16 +49,65 @@ We provide a number of optional filters you can use:
...
+	    'chargify_address_1'
+	    'chargify_address_2'
+	    'chargify_city'
+	    'chargify_state'
+	    'chargify_zip'
+	    'chargify_country'
+
 `chargify_account_details_fields` - Filter the Account Details form field. Use `$signup_form->add_field` and `$signup_form->remove_field` to add and/or remove fields.
+	Available fields by id;
+		'chargify_account_details_title'
```
Comment (BronsonQuick) [inline plugins/wp-chargify/readme.txt]: Another extra tab here.

**Review by BronsonQuick — CHANGES_REQUESTED** [review]
Awesome work dude! Just a couple of minor things. Thanks for fixing up some of the formatting too. I will have to check my sniffs on this new laptop as I don't think they are configured correctly yet.

File: plugins/wp-chargify/includes/forms/signup.php (hunk 2 above)
Comment (anthonythorne) [reply plugins/wp-chargify/includes/forms/signup.php]: this actually didn't work, I followed the cmb code back to 'form_format' where it was hardcoded, hence the form_format() function extending the hard code and allowing attributes, and or class name/extension from the original. Which came in handy during theme implementation.

I have removed the above line, which was remnant from testing.

File: plugins/wp-chargify/includes/forms/signup.php (hunk 1 above)
Comment (anthonythorne) [reply plugins/wp-chargify/includes/forms/signup.php]: Yep it does....

Removed it.

**Review by BronsonQuick — APPROVED** [review]
Looks good to me. Nice one!

**Pull request merged**